lib/TasmotaSerial-3.0.0/src/TasmotaSerial.cpp

```diff
@@ -174,6 +174,7 @@ bool TasmotaSerial::begin(long speed, int stop_bits) {
       m_uart = tasmota_serial_index;
       tasmota_serial_index--;
       TSerial = new HardwareSerial(m_uart);
+      if (TM_SERIAL_BUFFER_SIZE != serial_buffer_size) TSerial->setRxBufferSize(serial_buffer_size);
```

Comment:
Considering the default size of 256 for hardware serial I think we better change this line into ```
if (256 != serial_buffer_size) TSerial->setRxBufferSize(serial_buffer_size);
```